code/src/components/TaskList.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import moment from 'moment/moment';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { task } from 'reducers/task';
+
+export const TaskList = () => {
+  const tasklist = useSelector((store) => store.tasks.items);
+  const dispatch = useDispatch();
+
+  const onDeleteAllTasksClick = (id) => {
+    dispatch(task.actions.deleteSingleTask(id))
+  }
+  const onIsCompletedCheckboxToggle = (id) => {
+    dispatch(task.actions.toggleIfTaskIsCompleted(id))
+  }
+  const dateOptions = moment().format('ddd DD/M');
+
+  return (
+    <section>
+      <div className="header">
+        <h1>Grocery list</h1><p>{dateOptions}</p>
+      </div>
+      <ul>
+        {tasklist.map((singleTask) => {
+          return (
+            <li key={singleTask.id}>
+              <p>
+                <span>{singleTask.name}</span>
+                <label htmlFor={`task_with_id${singleTask.id}`} className="container">
+                  <input
+                    id={`task_with_id${singleTask.id}`}
+                    type="checkbox"
+                    value={singleTask.isCompleted}
+                    onChange={() => onIsCompletedCheckboxToggle(singleTask.id)} />
+                  <span className="checkmark" />
+                </label>
+                <button
+                  type="button"
+                  onClick={() => onDeleteAllTasksClick(singleTask.id)}
+                  className="remove-btn">✗
+                </button>
+              </p>
+            </li>)
+        })}
+      </ul>
+      <p className="total">Total: {tasklist.length}</p>
+    </section>
+  )
+}
```

Comment:
Interesting structure placing a ```<button>``` and ```<label>``` inside a ```<li>``` after wrapping it in a ```<p>```. I looked it up and it seems semantically correct!